eng/pipelines/templates/jobs/archetype-sdk-tests.yml

```diff
@@ -0,0 +1,114 @@
+
+    pool:
+      vmImage: "$(OSVmImage)"
+
+    timeoutInMinutes: 240
+
+    steps:
+      - task: NodeTool@0
+        inputs:
+          versionSpec: "$(NodeVersion)"
+        displayName: "Install Node.js $(NodeVersion)"
+
+      - script: |
+          node common/scripts/install-run-rush.js install
+        displayName: "Install dependencies"
+
+      - script: |
+          node common/scripts/install-run-rush.js build -t "${{parameters.PackageName}}" --verbose
+        displayName: 'rush build -t "${{parameters.PackageName}}"'
+
+      - script: |
+          node common/scripts/install-run-rush.js build:test -t "${{parameters.PackageName}}" --verbose
+        displayName: 'rush build:test -t "${{parameters.PackageName}}"'
+
+      - script: |
+          node common/scripts/install-run-rush.js integration-test:node -t "${{parameters.PackageName}}" --verbose
+        displayName: 'rush integration-test:node -t "${{parameters.PackageName}}"'
+        env:
+          ${{ each var in parameters.envvars }}:
+            ${{ var.key }}: ${{ var.value }}
+
+      - task: PublishTestResults@2
+        inputs:
+          testResultsFiles: "**/test-results.xml"
+          testRunTitle: "$(OSName) Node $(NodeVersion) - Node"
+        condition: succeededOrFailed()
+        displayName: "Publish node test results"
+
+  - job: "Test_Browser"
+    variables:
+      - template: ../variables/globals.yml
+
+    strategy:
+      # Service Bus and Event Hubs currently lack support for running multiple test suites in parallel,
+      # since the tests would attempt to use the same resources which causes conflicts.
+      maxParallel: 1
+      matrix:
+        Linux_Node10X:
+          OSName: "Linux"
+          OSVmImage: "ubuntu-16.04"
+        Windows_Node10X:
+          OSName: "Windows"
+          OSVmImage: "vs2017-win2016"
+        Mac_Node10X:
+          OSName: "Mac"
+          OSVmImage: "macOS-10.13"
+
+    pool:
+      vmImage: "$(OSVmImage)"
+
+    timeoutInMinutes: 240
+
+    steps:
+      - task: NodeTool@0
+        inputs:
+          versionSpec: "$(NodeVersion)"
+        displayName: "Install Node.js $(NodeVersion)"
+
+      - script: |
+          node common/scripts/install-run-rush.js install
+        displayName: "Install dependencies"
+
+      - script: |
+          node common/scripts/install-run-rush.js build -t "${{parameters.PackageName}}" --verbose
+        displayName: 'rush build -t "${{parameters.PackageName}}"'
+
+      - script: |
+          node common/scripts/install-run-rush.js build:test -t "${{parameters.PackageName}}" --verbose
+        displayName: 'rush build:test -t "${{parameters.PackageName}}"'
+
+      - script: |
+          node common/scripts/install-run-rush.js integration-test:browser -t "${{parameters.PackageName}}" --verbose
+        condition: succeededOrFailed()
+        displayName: 'rush integration-test:browser -t "${{parameters.PackageName}}"'
+        env:
+          ${{ each var in parameters.envvars }}:
```

Comment:
I'm not 100% certain but you might need to define the envvars parameter at the top of this template so that if it is ever not provided this can still correctly evaluate.